a2wsgi/asgi.py

```diff
@@ -203,16 +203,29 @@ def start_asgi_app(self, environ: Environ) -> asyncio.Task:
         run_asgi.add_done_callback(self.asgi_done_callback)
         return run_asgi
 
+    def call_in_loop(self, func, *args):
+        """Call func(*args) in the asyncio loop, in a thread-safe way.
+
+        Return the result.
+        """
+        # Asyncio's call_soon_threadsafe doesn't return the result;
+        # to get that we need to use run_coroutine_threadsafe, which
+        # requires a coroutine (async function).
+        async def coro():
+            return func(*args)
+        future = asyncio.run_coroutine_threadsafe(coro(), self.loop)
+        # Block until the result is available, and return it.
+        return future.result()
+
     def __call__(
         self, environ: Environ, start_response: StartResponse
     ) -> IterableChunks:
         read_count: int = 0
         body = environ["wsgi.input"] or BytesIO()
         content_length = int(environ.get("CONTENT_LENGTH", None) or 0)
 
-        asgi_task = self.start_asgi_app(environ)
         # activate loop
-        self.loop.call_soon_threadsafe(lambda: None)
+        asgi_task = self.call_in_loop(self.start_asgi_app, environ)
```

Comment:
You did not understand the original purpose here. Thank you for your enthusiasm, but please learn how to write asyncio code first.